### PR TITLE
BcmcCardHolder: Add card holder name input to bcmc

### PR DESCRIPTION
**Pull request by OscarSpruit**

### CODE REVIEW

File: bcmc/src/main/java/com/adyen/checkout/bcmc/BcmcView.kt [hunk 1]
```diff
@@ -111,6 +117,24 @@ internal class BcmcView @JvmOverloads constructor(
...
+        binding.editTextCardHolder.setOnChangeListener {
+            delegate.updateInputData { cardHolderName = binding.editTextCardHolder.rawValue }
+            binding.textInputLayoutCardHolder.hideError()
+        }
+
+        binding.editTextCardHolder.onFocusChangeListener = OnFocusChangeListener { _, hasFocus ->
+            val cardHolderValidation = delegate.outputData.cardHolderNameField.validation
+            if (hasFocus) {
+                binding.textInputLayoutCardHolder.hideError()
+            } else if (!cardHolderValidation.isValid()) {
```
Comment (ozgur00) [inline bcmc/src/main/java/com/adyen/checkout/bcmc/BcmcView.kt:131]: we can use `cardHolderValidation is Validation.Invalid` so we won't have to cast it below. there are examples in `CardView`. I think this PR is a good place to refactor all the other places that uses `isValid()` for this purpose.

Comment (OscarSpruit) [reply bcmc/src/main/java/com/adyen/checkout/bcmc/BcmcView.kt:131]: Good one! I indeed got this from `CardView`, so will change it everywhere I can find

**Review by ozgur00 — APPROVED** [review]

File: bcmc/src/main/java/com/adyen/checkout/bcmc/BcmcConfiguration.kt [hunk 2]
```diff
@@ -98,6 +106,17 @@ class BcmcConfiguration : Configuration {
             return super.setEnvironment(environment) as Builder
         }
 
+        /**
+         * Set if the holder name is required and should be shown as an input field.
+         *
+         * @param isHolderNameRequired [Boolean]
+         * @return [CardConfiguration.Builder]
```
Comment (jreij) [inline bcmc/src/main/java/com/adyen/checkout/bcmc/BcmcConfiguration.kt:113]: Should be `BcmcConfiguration.Builder`

File: bcmc/src/main/java/com/adyen/checkout/bcmc/BcmcView.kt [hunk 3]
```diff
@@ -70,6 +72,10 @@ internal class BcmcView @JvmOverloads constructor(
                 R.style.AdyenCheckout_Card_ExpiryDateInput,
                 localizedContext
             )
+            binding.textInputLayoutCardHolder.setLocalizedHintFromStyle(
+                com.adyen.checkout.card.R.style.AdyenCheckout_Card_HolderNameInput,
```
Comment (jreij) [inline bcmc/src/main/java/com/adyen/checkout/bcmc/BcmcView.kt:76]: Do we need the full `com.adyen.checkout.card.R` or can we just use `R`?

File: example-app/src/main/java/com/adyen/checkout/example/ui/configuration/CheckoutConfigurationProvider.kt [hunk 4]
```diff
@@ -74,6 +74,7 @@ internal class CheckoutConfigurationProvider @Inject constructor(
     private fun getBcmcConfiguration(): BcmcConfiguration =
         BcmcConfiguration.Builder(shopperLocale, environment, clientKey)
             .setShopperReference(keyValueStorage.getShopperReference())
+            .setHolderNameRequired(true)
```
Comment (jreij) [inline example-app/src/main/java/com/adyen/checkout/example/ui/configuration/CheckoutConfigurationProvider.kt:77]: Maybe let's not enable it by default in the example app, to be consistent with card

**Pull request merged**